docs/content/preview/explore/query-1-performance/pg-stats.md

```diff
@@ -0,0 +1,143 @@
+```
+
+## Fetching the statistics
+
+Once the ANALYZE command is run, the `pg_statistic` command is populated with the statistics of the data. These stats are made easily readbable via the pg_stats view. For example, you can view the statistics of the `name` column in the `users` table as,
+
+```sql
+SELECT attname, null_frac, n_distinct, most_common_vals, most_common_freqs
+    FROM pg_stats
+    WHERE tablename='users' AND attname='name';
+```
+
+You should get an output similar to:
+
+```caddyfile{.nocopy}
+-[ RECORD 1 ]-----+------------------------------------------------------------------------
+attname           | name
+null_frac         | 0.4042
+n_distinct        | 10
+most_common_vals  | {"",Jordan,Jamie,Morgan,Taylor,Kim,Pat,Casey,Sam,Lee}
+most_common_freqs | {0.2963,0.0352,0.0346,0.0342,0.0339,0.0334,0.0333,0.0328,0.0313,0.0308}
+```
+
+The above output says that,
+
+- **null_frac** shows that the fraction of null values is about 40%. (_Remember_, we tried to insert about 40% NULL values)
+- **n_distinct** shows that there are about 10 distinct values in the name column
+- **most_common_vals** shows the most commonly occuring values (other than NULL)
+- **most_common_freqs** shows the frequency in which the most common values occur. For example, Empty value occurs at 29.6%, John occurs at 3.0% and so on.
+
+## Partial indexes
+
+Now when we try to create an index on the column `name`, the index could be unevenly distributed. This is due to 40% of the dataset consisting of NULL values and 30% consisting of empty values. Only 30% consist valid values. If we know that our queries will not lookup NULL or empty values, we can create partial index as,
+
+```sql
+CREATE INDEX idx_users_name_nonempty
+    ON users (name)
+    WHERE name IS NOT NULL AND name <> '';
+```
+
+The above index will have just the valid values and the index will not be skewed.
+
+## Cardinality
+
+Given that there are only 9 (via **n_distinct**) distinct valid (without empty string) values in the `name` column, on an index with HASH distribution, even if there are more nodes in the cluster, the index will end up being only on a maximum of 9 nodes. This is because each value is hashed to determine the tablet in which it should be stored and there will be only 9 distinct hashes. This is the reason why is not advisable to create indexes on low cardinality columns like Booleans (2), Days of week(7) etc.
+
+## Skewed data
+
+It is always advisable for your index/table to be reasonably distributed so that all nodes in your cluster handle a reasonably equal share of queries. In the above data set, it is easy to see that together `NULL` and empty values account for 70% of the total data set (via **most_common_vals, null_frac**). And the valid names constitute only 30% of the dataset. If an index was created will all these values then all queries to the index will be handled only by about 30% of the nodes in the cluster as 70% of data would not be queried against.
```

Comment:
```suggestion
It is always advisable for your index/table to be reasonably distributed so that all nodes in your cluster process a similar amount of queries. In the above data set, `NULL` and empty values make up 70% of the total dataset (via **most_common_vals, null_frac**), with valid names only accounting for 30%.  If you create an index on this data, only approximately 30% of the nodes will handle most of the queries, as 70% of the data is not queried.
```